SU2_CFD/src/fluid/CFluidScalar.cpp

```diff
@@ -0,0 +1,245 @@
+  su2double mixtureFractionDenumerator = 0.0;
+  const su2double A = 0.375;
+  std::vector<su2double> mixtureFractions;
+  mixtureFractions.clear();
+
+  for (int iVar = 0; iVar < n_species_mixture; iVar++){
+    LaminarViscosityPointers[iVar]->SetViscosity(Temperature, Density);
+    laminarViscosity.at(iVar) = LaminarViscosityPointers[iVar]->GetViscosity();
+  }
+
+  for(int i = 0; i < n_species_mixture; i++){
+    mixtureFractionDenumerator += moleFractions[i] * sqrt(molarMasses[i]);
+  }
+
+  for(int j = 0; j < n_species_mixture; j++){
+    mixtureFractions.push_back((moleFractions[j] * sqrt(molarMasses[j])) / mixtureFractionDenumerator);
+  }
+
+  for(int i = 0; i < n_species_mixture; i++){
+    for(int j = 0; j < n_species_mixture; j++){
+      E = (2*sqrt(molarMasses[i]) * sqrt(molarMasses[j])) / (molarMasses[i] + molarMasses[j]);
+      fluidity += ((mixtureFractions[i] * mixtureFractions[j]) / (sqrt(laminarViscosity[i]) * sqrt(laminarViscosity[j]))) * pow(E, A);
+    }
+  }
+  return viscosityMixture = 1 / fluidity;
+}
+
+su2double CFluidScalar::wilkeConductivity(const su2double * const val_scalars){
+  std::vector<su2double> phi;
+  std::vector<su2double> wilkeNumerator;
+  std::vector<su2double> wilkeDenumeratorSum;
+  su2double wilkeDenumerator = 0.0;
+  su2double conductivityMixture = 0.0;
+  wilkeDenumeratorSum.clear();
+  wilkeNumerator.clear();
+
+  for (int iVar = 0; iVar < n_species_mixture; iVar++){
+    ThermalConductivityPointers[iVar]->SetConductivity(Temperature, Density, Mu, Mu_Turb, Cp);
+    laminarthermalConductivity.at(iVar) = ThermalConductivityPointers[iVar]->GetConductivity();
+  }
+
+  for(int i = 0; i < n_species_mixture; i++){
+    for(int j = 0; j < n_species_mixture; j++){
+      phi.push_back(pow((1 + pow((laminarViscosity[i]) / (laminarViscosity[j]),0.5) * pow(molarMasses[j] / molarMasses[i],0.25)),2) / pow(8 * (1 + molarMasses[i] / molarMasses[j]),0.5));
+      wilkeDenumerator += moleFractions[j] * phi[j];
+    }
+    wilkeDenumeratorSum.push_back(wilkeDenumerator);
+    wilkeDenumerator = 0.0;
+    phi.clear();
+    wilkeNumerator.push_back(moleFractions[i] * laminarthermalConductivity[i]);
+    conductivityMixture += wilkeNumerator[i] / wilkeDenumeratorSum[i];
+  }
+  return conductivityMixture;
+}
+
+unsigned long CFluidScalar::SetTDState_T(const su2double val_temperature, su2double * const val_scalars){
+  MeanMolecularWeight = 1/(val_scalars[0]/(molarMasses[0]/1000) + (1-val_scalars[0])/(molarMasses[1]/1000));
+
+  // Cp = specificHeat[0] * val_scalars[0] + specificHeat[1] * (1- val_scalars[0]);
+  Cp = 1009.39;
+  // Cp = 1009.3;
+  Cv = Cp/1.4;
+  Temperature = val_temperature;
+  Density = Pressure / ((Temperature * UNIVERSAL_GAS_CONSTANT) / MeanMolecularWeight);
+
+  massToMoleFractions(val_scalars);
+
+  if(wilke){
+    Mu  = wilkeViscosity(val_scalars);
+  }
+  else if(davidson){
+    Mu = davidsonViscosity(val_scalars);
+  }
+
+  Kt = wilkeConductivity(val_scalars);
+
+  return 0;
```

Comment:
I guess you didn't write the function prototype but returning sth without use is not good. But if it comes from feature_flamelet do not change